include/mlir/IR/OpImplementation.h

```diff
@@ -410,10 +410,25 @@ class OpAsmParser {
                                           ArrayRef<OperandType> arguments,
                                           ArrayRef<Type> argTypes) = 0;
 
-  /// Parse a region argument.  Region arguments define new values, so this also
-  /// checks if the values with the same name has not been defined yet.
+  /// Parse a region argument.  Region arguments define new values; so this also
+  /// checks if values with the same name have not been defined yet.
   virtual ParseResult parseRegionArgument(OperandType &argument) = 0;
 
+  /// Parse zero or more region arguments with a specified surrounding
+  /// delimiter, and an optional required argument count. Region arguments
+  /// define new values; so this also checks if values with the same names have
+  /// not been defined yet.
+  virtual ParseResult
+  parseRegionArgumentList(SmallVectorImpl<OperandType> &result,
+                          int requiredOperandCount = -1,
+                          Delimiter delimiter = Delimiter::None) = 0;
+  virtual ParseResult
+  parseRegionArgumentList(SmallVectorImpl<OperandType> &result,
+                          Delimiter delimiter) {
```

Comment:
I'd prefer to not have this overload, it doesn't add much. If useful, we could interchange delimiter and requiredOperandCount based on usage and that would then cover this case.

Reply:
I was just following parseOperandList, which is ditto in spirit, and has this overload. (If we don't want this, we should remove it there too to be consistent IMO. Let me know.)

Reply:
That could be done in a follow up yes, not along with this.

Reply:
Thanks.